homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -121,7 +176,23 @@ def name(self):
     @property
     def state(self):
         """Return the state of the sensor."""
-        return self._state
+        if self._type != "inverters":
+            value = self.coordinator.data.get(self._type)
+
+        elif (
+            self._type == "inverters"
+            and self.coordinator.data.get("inverters_production") is not None
+        ):
+            split_name = self._name.split(" ")
+            serial_number = split_name[-1]
```

Comment:
Pass the serial number as a parameter to the entity.

Reply:
I grab the serial number when the array is created that is passed to `async_add_entities`, which is than passed as a parameter to the entity